src/main/java/com/microsoft/graph/auth/BaseAuthentication.java

```diff
@@ -0,0 +1,96 @@
+package com.microsoft.graph.auth;
+
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.oltu.oauth2.client.OAuthClient;
+import org.apache.oltu.oauth2.client.URLConnectionClient;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest.TokenRequestBuilder;
+import org.apache.oltu.oauth2.client.response.OAuthJSONAccessTokenResponse;
+import org.apache.oltu.oauth2.common.message.types.GrantType;
+
+import com.microsoft.graph.auth.enums.NationalCloud;
+
+public class BaseAuthentication {
+
+	protected List<String> Scopes;
+	protected String ClientId;
+	protected String authority;
+	protected String ClientSecret;
+	protected long startTime;
+	protected NationalCloud nationalCloud;
+	protected String tenant;
+	protected String redirectUri = "https://localhost:8080";
+	protected OAuthJSONAccessTokenResponse response;
+
+	public BaseAuthentication(
+			List<String> scopes,
+			String clientId,
+			String authority,
+			String redirectUri,
+			NationalCloud nationalCloud,
+			String tenant,
+			String ClientSecret)
+	{
+		this.Scopes = scopes;
+		this.ClientId = clientId;
+		this.authority = authority;
+		this.redirectUri = redirectUri;
+		this.nationalCloud = nationalCloud;
+		this.tenant = tenant;
+		this.ClientSecret = ClientSecret;
+	}
+
+	protected static HashMap<String, String> CloudList = new HashMap<String, String>()
+	{{
+		put( "Global", "https://login.microsoftonline.com/" );
+		put( "China", "https://login.chinacloudapi.cn/" );
+		put( "Germany", "https://login.microsoftonline.de/" );
+		put( "UsGovernment", "https://login.microsoftonline.us/" );
+	}};
+
+	protected static String GetAuthority(NationalCloud authorityEndpoint, String tenant)
+	{
+		return CloudList.get(authorityEndpoint.toString()) + tenant;
+	}
+
+	protected String getScopesAsString() {
+		String scopeString = new String();
+		for(String s : this.Scopes) {
```

Comment:
you may want to use StringBuilder to construct string.

Reply:
Resolved this comment by using StringBuilder

Reply:
Resolved this by using StringBuilder.